### PR TITLE
blog v4 release

### PR DESCRIPTION
**Pull request by p-a-a-a-trick**
- docs: added 4.0 release blog post
- docs: update
- docs: update

### CODE REVIEW

File: docs/blog/ibis-version-4.0.0-release.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## Backends
+
+Ibis 4.0 brings [Polars](https://ibis-project.org/docs/latest/backends/Polars/), [Snowflake](https://ibis-project.org/docs/dev/backends/Snowflake/), and [Trino](https://ibis-project.org/docs/dev/backends/Trino/) into and already-impressive stock of supported backends.
+The [Polars](https://www.pola.rs/) backend adds another way for users to work locally with DataFrames, and the [Snowflake](https://www.snowflake.com/en/) and [Trino](https://trino.io/) backends adds a free and familiar python API to popular data warehouses.
+
+## Functionality
+
+There are a lot of improvements incoming, but some notable changes include:
+
+- [read](https://github.com/ibis-project/ibis/pull/4759) API ([including explicit functions](https://github.com/ibis-project/ibis/pull/5005)): allows users to read various file formats directly into their [configured `default_backend`](https://ibis-project.org/docs/dev/api/config/?h=default#ibis.config.Options) (default DuckDB).
```
Comment (jcrist) [inline docs/blog/ibis-version-4.0.0-release.md:23]: `read` has been split into `read_csv` and `read_parquet`, `ibis.read` no longer exists.

File: docs/blog/ibis-version-4.0.0-release.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- [to_pyarrow](https://github.com/ibis-project/ibis/pull/4454#issuecomment-1262640204): users can now return PyArrow objects (Tables, Arrays, Scalars) and therefore grants all of the functionality that PyArrow provides
```
Comment (jcrist) [inline docs/blog/ibis-version-4.0.0-release.md:24]: May also want to include `to_pyarrow_batches`, which returns a pyarrow `RecordBatchReader` for streaming result sets.

**Review by jcrist — COMMENTED** [review]
A couple small comments, but otherwise :+1:. Nice concise release blogpost.

File: docs/blog/ibis-version-4.0.0-release.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Ibis 4.0 has officially been released as the latest version of the package.
+With this release comes of more backends, improved functionality, several refactors, bug fixes, and more.
+A full list of the changes can be found in the project release notes [here](../release_notes.md).
+Let’s talk about some of the new changes 4.0 brings for Ibis users.
+
+## Backends
+
+Ibis 4.0 brings [Polars](https://ibis-project.org/docs/latest/backends/Polars/), [Snowflake](https://ibis-project.org/docs/dev/backends/Snowflake/), and [Trino](https://ibis-project.org/docs/dev/backends/Trino/) into and already-impressive stock of supported backends.
+The [Polars](https://www.pola.rs/) backend adds another way for users to work locally with DataFrames, and the [Snowflake](https://www.snowflake.com/en/) and [Trino](https://trino.io/) backends adds a free and familiar python API to popular data warehouses.
```
[the remark this answers is not in the document]
Comment (p-a-a-a-trick) [reply docs/blog/ibis-version-4.0.0-release.md]: [updated](https://github.com/ibis-project/ibis/pull/5177/files#diff-2ec843e97605581ceada30571b1090aabe0bf588427981f61eceb5086dfb1c9bR20)

File: docs/blog/ibis-version-4.0.0-release.md (hunk 1 above)
Comment (p-a-a-a-trick) [reply docs/blog/ibis-version-4.0.0-release.md]: [updated](https://github.com/ibis-project/ibis/pull/5177/files#diff-2ec843e97605581ceada30571b1090aabe0bf588427981f61eceb5086dfb1c9bR26)

File: docs/blog/ibis-version-4.0.0-release.md (hunk 2 above)
Comment (p-a-a-a-trick) [reply docs/blog/ibis-version-4.0.0-release.md]: [updated](https://github.com/ibis-project/ibis/pull/5177/files#diff-2ec843e97605581ceada30571b1090aabe0bf588427981f61eceb5086dfb1c9bR27)

File: docs/blog/ibis-version-4.0.0-release.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+## Introduction
+
+Ibis 4.0 has officially been released as the latest version of the package.
+This release includes several new backends, improved functionality, and some major internal refactors.
+A full list of the changes can be found in the project release notes [here](../release_notes.md).
+Let’s talk about some of the new changes 4.0 brings for Ibis users.
+
+## Backends
+
+Ibis 4.0 brings [Polars](https://ibis-project.org/docs/latest/backends/Polars/), [Snowflake](https://ibis-project.org/docs/dev/backends/Snowflake/), and [Trino](https://ibis-project.org/docs/dev/backends/Trino/) into and already-impressive stock of supported backends.
```
Comment (gforsyth) [inline docs/blog/ibis-version-4.0.0-release.md:16]:
```suggestion
Ibis 4.0 brings [Polars](https://ibis-project.org/docs/latest/backends/Polars/), [Snowflake](https://ibis-project.org/docs/dev/backends/Snowflake/), and [Trino](https://ibis-project.org/docs/dev/backends/Trino/) into an already-impressive stock of supported backends.
```

**Review by gforsyth — COMMENTED** [review]
micronit -- overall looks good!

File: docs/blog/ibis-version-4.0.0-release.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+A full list of the changes can be found in the project release notes [here](../release_notes.md).
+Let’s talk about some of the new changes 4.0 brings for Ibis users.
+
+## Backends
+
+Ibis 4.0 brings [Polars](https://ibis-project.org/docs/latest/backends/Polars/), [Snowflake](https://ibis-project.org/docs/dev/backends/Snowflake/), and [Trino](https://ibis-project.org/docs/dev/backends/Trino/) into an already-impressive stock of supported backends.
+The [Polars](https://www.pola.rs/) backend adds another way for users to work locally with DataFrames.
+The [Snowflake](https://www.snowflake.com/en/) and [Trino](https://trino.io/) backends add a free and familiar python API to popular data warehouses.
+
+Alongside these new backends, Google's BigQuery and mssql have been moved to the main repo and have been updated.
```
Comment (krzysztof-kwitt) [inline docs/blog/ibis-version-4.0.0-release.md:20]:
```suggestion
Alongside these new backends, Google BigQuery and Microsoft SQL have been moved to the main repo and have been updated.
```

File: docs/blog/ibis-version-4.0.0-release.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+**by Patrick Clarke**
+
+09 January 2023
+
+## Introduction
+
+Ibis 4.0 has officially been released as the latest version of the package.
+This release includes several new backends, improved functionality, and some major internal refactors.
+A full list of the changes can be found in the project release notes [here](../release_notes.md).
```
Comment (krzysztof-kwitt) [inline docs/blog/ibis-version-4.0.0-release.md:11]: Can you avoid using "here" word in link?

> Write unique, descriptive link text that makes sense without the surrounding text. Don't use phrases such as this document, this article, or click here.


https://developers.google.com/style/link-text

File: docs/blog/ibis-version-4.0.0-release.md (hunk 5 above)
Comment (krzysztof-kwitt) [inline docs/blog/ibis-version-4.0.0-release.md]: What's the benefit of moving code to this repository? I understand the benefit it gives, but I don't know if the user who reads this text will also know it.

File: docs/blog/ibis-version-4.0.0-release.md (hunk 6 above)
Comment (cpcloud) [inline docs/blog/ibis-version-4.0.0-release.md:11]:
```suggestion
A full list of the changes can be found in the [project release notes](../release_notes.md).
```

**Review by cpcloud — APPROVED** [review]

**Pull request merged**